dali/kernels/imgproc/color_manipulation/brightness_contrast_gpu_test.cu

```diff
@@ -0,0 +1,149 @@
+// Copyright (c) 2019, NVIDIA CORPORATION. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <gtest/gtest.h>
+#include <vector>
+#include <tuple>
+#include "dali/kernels/test/tensor_test_utils.h"
+#include "dali/kernels/test/kernel_test_utils.h"
+#include "dali/kernels/imgproc/color_manipulation/brightness_contrast_gpu.cuh"
+#include "dali/kernels/imgproc/color_manipulation/brightness_contrast_test_utils.h"
+
+namespace dali {
+namespace kernels {
+namespace brightness_contrast {
+namespace test {
+
+namespace {
+
+template <class In, class Out>
+std::enable_if_t<std::is_integral<Out>::value, Out> custom_round(float val) {
+  return std::round(val);//TODO round to nearest even
+}
+
+
+template <class In, class Out>
+std::enable_if_t<!std::is_integral<Out>::value, Out> custom_round(float val) {
+  return val;
+}
+
+
+}  // namespace
+
+template <class InputOutputTypes>
+class BrightnessContrastCudaKernelTest : public ::testing::Test {
+  using In = typename InputOutputTypes::In;
+  using Out = typename InputOutputTypes::Out;
+
+ protected:
+  BrightnessContrastCudaKernelTest() {
+    input_host_.resize(dataset_size());
+  }
+
+
+  void SetUp() final {
+    std::mt19937_64 rng;
+    UniformRandomFill(input_host_, rng, 0., 10.);
+    calc_output();
+    CUDA_CALL(cudaMalloc(&input_device_, sizeof(In) * dataset_size()));
+    CUDA_CALL(cudaMemcpy(input_device_, input_host_.data(), input_host_.size() * sizeof(In),
+                         cudaMemcpyDefault));
+    CUDA_CALL(cudaMallocManaged(&output_, dataset_size() * sizeof(Out)));
```

Comment:
We have problems with the resize op and devices not supporting unified memory. I would recommend avoiding it here as well or providing a second code path for such configuration.